### ISSUE TITLE
Adjusting optimizer learning rates on-the-fly

### ISSUE THREAD

**Issue by staticfloat** [FluxML/Flux.jl#233]
It would be great to allow the Flux optimizers to have parameters that can be tweaked from the outside.  As an example, I would like to have an SGD where I can tune the learning rate over time.  I could do this by destroying the optimizer and recreating it every epoch, (which is fine for SGD) but that hurts performance with optimizers that keep running estimates of gradient magnitude such as ADAM or RMSProp.

**Comment by tejank10** [comment]
I am thinking of treating the optimizers as `struct`s rather than `function`s. For example, `ADAM` will be a `struct` having learning rate, momentum, velocity etc. as its fields. And there can be an `update` function which takes this `struct` and performs the update, thereby updating the momentum and velocity.

**Comment by iblis17** [comment]
Currently, I tweak learning rate via opening the training loop and change the `Δ` of `back!(loss, Δ)`.

**Comment by MikeInnes** [comment]
Fixed by #379.
